src/plugins/controls/public/control_group/editor/control_editor.tsx

```diff
@@ -88,59 +143,54 @@ export const ControlEditor = ({
       </EuiFlyoutHeader>
       <EuiFlyoutBody data-test-subj="control-editor-flyout">
         <EuiForm>
-          <EuiSpacer size="l" />
-          {ControlTypeEditor && (
-            <ControlTypeEditor
-              onChange={onTypeEditorChange}
-              setValidState={setControlEditorValid}
-              initialInput={embeddable?.getInput()}
-              setDefaultTitle={(newDefaultTitle) => {
-                if (!currentTitle || currentTitle === defaultTitle) {
-                  setCurrentTitle(newDefaultTitle);
-                  updateTitle(newDefaultTitle);
-                }
-                setDefaultTitle(newDefaultTitle);
-              }}
-            />
-          )}
-          <EuiFormRow label={ControlGroupStrings.manageControl.getTitleInputTitle()}>
-            <EuiFieldText
-              data-test-subj="control-editor-title-input"
-              placeholder={defaultTitle}
-              value={currentTitle}
-              onChange={(e) => {
-                updateTitle(e.target.value || defaultTitle);
-                setCurrentTitle(e.target.value);
-              }}
-            />
-          </EuiFormRow>
-          <EuiFormRow label={ControlGroupStrings.manageControl.getWidthInputTitle()}>
-            <EuiButtonGroup
-              color="primary"
-              legend={ControlGroupStrings.management.controlWidth.getWidthSwitchLegend()}
-              options={CONTROL_WIDTH_OPTIONS}
-              idSelected={currentWidth}
-              onChange={(newWidth: string) => {
-                setCurrentWidth(newWidth as ControlWidth);
-                updateWidth(newWidth as ControlWidth);
-              }}
-            />
+          <EuiFormRow label={ControlGroupStrings.manageControl.getControlTypeTitle()}>
+            <EuiKeyPadMenu>
+              {isCreate ? getTypeButtons(getControlTypes()) : getTypeButtons([selectedType])}
+            </EuiKeyPadMenu>
           </EuiFormRow>
-          <EuiSpacer size="l" />
-          {removeControl && (
-            <EuiButtonEmpty
-              aria-label={`delete-${title}`}
-              iconType="trash"
-              flush="left"
-              color="danger"
-              onClick={() => {
-                onCancel();
-                removeControl();
-              }}
-            >
-              {ControlGroupStrings.management.getDeleteButtonTitle()}
-            </EuiButtonEmpty>
-          )}
+          {selectedType ? (
```

Comment:
Nit: A way to accomplish this a bit nicer is to use:

```suggestion
          {selectedType && (
```

then you can get rid of the `: null` at the end. It's best not to use a ternary if not necessary!